dts/bindings/iio/adc/atmel,sam0-adc.yaml

```diff
@@ -23,14 +23,8 @@ properties:
     interrupts:
       category: required
 
-    gclk:
+    clock-frequency:
```

Comment:
Why isn't the clock-freq figured out from the clock for this device?

Reply:
Similar to the `counter`, the ADC has an on-peripheral divisor available, which this is used to calculate.  By default, this is set to the maximum frequency the ADC can support (determined by hardware, e.g. 2.1 MHz on SAMD2x or 16 MHz on SAME54), so this is used simply to calculate the divisor needed to keep the ADC within spec. 
 But it can also be used to apply the divisor to achieve different conversion rates (e.g. setting it to 1 MHz to allow for longer sample acquisition times).

I have clarified the description on the binding.